clock_control: stm32: F4 domain clock + fixed domain clock

erwango opened the pull request:
This PR contains 2 changes required for https://github.com/zephyrproject-rtos/zephyr/pull/53258

1- Provide DCKCFGR description on F4 series
2- Add ability to describe in dts fixed domain clocks

erwango committed:
include: dt-bindings: stm32f4_clock.h: Add DCKCFGR registers

Add f4 binding files to provide DCKCFGR registers description
to enable clock selection for F410/F427/F446 variant lines

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented on dts/arm/st/f4/stm32f469.dtsi:
```diff
@@ -10,6 +10,8 @@
 		usbotg_fs: usb@50000000 {
 			num-bidir-endpoints = <6>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00000080>,
+				 <&rcc STM32_SRC_PLL_Q CLK48M_SEL(0)>;
```
Remove, thanks

FRASTM commented on drivers/clock_control/clock_stm32_ll_common.c:
```diff
@@ -242,6 +242,11 @@ static inline int stm32_clock_control_configure(const struct device *dev,
 		return err;
 	}
 
+	if (pclken->enr == NO_SEL) {
```
What is a usecase where pclken->enr is NO_SEL is it a "default value"

erwango replied:
See commit description:
 USB clock on PLL_Q output on F405 devices is not selectable. But yet it needs to be described in device tree in order to be able to call `clock_control_get_rate`.

FRASTM commented on include/zephyr/dt-bindings/clock/stm32f0_clock.h:
```diff
@@ -73,5 +73,7 @@
 /** BDCR devices */
 #define RTC_SEL(val)		STM32_CLOCK(val, 3, 8, BDCR_REG)
+/** Dummy: Add a specificier when no selection is possible */
+#define NO_SEL			0xFF
```
could   `NO_SEL`  be defined only _once_ in the #include <zephyr/drivers/clock_control/stm32_clock_control.h>

erwango replied:
Not possible. `drivers/clock_control/stm32_clock_control.h` is not a device tree binding (could not be included in *.dtsi)

FRASTM commented on dts/arm/st/f4/stm32f410.dtsi:
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/dt-bindings/clock/stm32f4_clock.h>
```
See that some other stm32f4x devices have a DCKCFGR2 : stm32f410, f412, f413, f423

erwango committed:
dts: stm32f4: stm32f412 is a variant of stm32f410

In order to ease description of DCKCFG regsiters,
make f412 a variant of f410 as it supposed to be.

Only exception is missing DAC1.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts: stm32f4: rng is available on stm32f410

Add rng definition to f410.
Though, don't inherit directly in f412 as it's integrated
in a different way.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
dts: stm32f4: stm32f437 is a variant of stm32f427

stm32f427 + crypto = stm32f437

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
drivers: clock_control: stm32: Fixed domain clock configuration

In some case, we may need to describe a domain clock for a device
while there is no way to configure it (ex: USB clock set on PLL_Q output
on F405 devices > It is not selectable).
Then, configuring a device clock domain in the clock_control driver
will allow to retrieve its subsys rate.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

gautierg-st reviewed (changes requested):
Apart from the comments, LGTM

FRASTM commented on dts/arm/st/f4/stm32f410.dtsi:
```diff
@@ -91,6 +96,13 @@
+			compatible = "st,stm32-rng";
+			reg = <0x50060800 0x400>;
+			interrupts = <80 0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00000040>;
```
Bit 31 RNGEN: RNG clock enable of the RCC AHB1 peripheral clock enable register : RCC_AHB1ENR

erwango commented on dts/arm/st/f4/stm32f437.dtsi:
```diff
@@ -4,97 +4,11 @@
-
-		/* spi5 is present on all STM32F437XX SoCs except
-		 * STM32F437vX SoCs. Delete node in stm32f437vX.dtsi.
-		 */
```
Fixed

erwango commented on dts/arm/st/f4/stm32f437.dtsi:
```diff
@@ -4,97 +4,11 @@
-
-		/* spi6 is present on all STM32F437XX SoCs except
-		 * STM32F437vX SoCs. Delete node in stm32f437vX.dtsi.
-		 */
```
Fixed
